MediaBrowser.Api/Playback/Hls/DynamicHlsService.cs

```diff
@@ -927,35 +927,41 @@ protected override string GetVideoArguments(StreamState state, EncodingOptions e
             }
             else
             {
+                var gopArg = string.Empty;
                 var keyFrameArg = string.Format(
                     CultureInfo.InvariantCulture,
                     " -force_key_frames:0 \"expr:gte(t,{0}+n_forced*{1})\"",
                     GetStartNumber(state) * state.SegmentLength,
                     state.SegmentLength);
-                if (state.TargetFramerate.HasValue)
+
+                var framerate = (state.VideoStream != null) ? state.VideoStream.RealFrameRate : null;
+
+                if (framerate.HasValue)
```

Comment:
Won't this fail when `framerate` is `null`? I'm not proficient enough in C# to tell without testing...

Reply:
Generally this value is not empty unless `ffprobe` does not detect the video framerate when scanning the library. However, it is better to modify it for caution.

Reply:
Well, I just googled and `framerate.HasValue` shouldn't fire if `framerate` is of type `float?`, but given that `var framerate` declaration I'm not sure it is.